### PR TITLE
Added possibility to change emails of user

### PR DESCRIPTION
**Pull request by azomazo**

### CODE REVIEW

File: lib/gitlab/client/emails.rb [hunk 1]
```diff
@@ -0,0 +1,60 @@
+class Gitlab::Client
+  # Defines methods related to user emails.
+  # @see https://github.com/gitlabhq/gitlabhq/blob/master/doc/api/users.md
+  module Emails
```
Comment (NARKOZ) [inline lib/gitlab/client/emails.rb]: Put methods into existing Users module.

File: lib/gitlab/client/emails.rb [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    # @example
+    #   Gitlab.delete_email(2)
+    #   Gitlab.delete_email(3, 2)
+    #
+    # @param  [Integer] id Email address ID
+    # @param  [Integer] user_id The ID of a user.
+    # @return [Boolean]
+    def delete_email(id, user_id=nil)
+      url = user_id.to_i.zero? ? "/user/emails/#{id}" : "/users/#{user_id}/emails/#{id}"
+      delete(url, parser: ->(body, _) { body == 'null' ? true :  Gitlab::Request.parse(body) })
```
Comment (NARKOZ) [inline lib/gitlab/client/emails.rb]: Parsing should be handled in [Request class](https://github.com/NARKOZ/gitlab/blob/master/lib%2Fgitlab%2Frequest.rb#L15-L15)

**Comment by azomazo** [comment]
Moved code Users and remove override parser in `delete_email` method.
But method `delete_email` always returns `null`. Therefore, the result of the method turns out false, which can be misleading.

**Comment by NARKOZ** [comment]
Thank you.

>Therefore, the result of the method turns out false, which can be misleading.

Yes, but that's because of the GitLab API's inconsistency.

**Pull request merged**